fetching/tarball-fetcher/src/gitHostedTarballFetcher.ts

```diff
@@ -49,17 +54,32 @@ async function prepareGitHostedPkg (
     },
     force: true,
   })
-  const shouldBeBuilt = await preparePackage(opts, tempLocation)
+  const shouldBeBuilt = await preparePackage(opts, tempLocation, resolution.path ?? tempLocation)
   // Important! We cannot remove the temp location at this stage.
   // Even though we have the index of the package,
   // the linking of files to the store is in progress.
   return {
     ...await addFilesFromDir({
       cafsDir: cafs.cafsDir,
-      dir: tempLocation,
+      dir: resolution.path
+        ? getJoinedPath(tempLocation, resolution.path)
+        : tempLocation,
       filesIndexFile,
       pkg: fetcherOpts.pkg,
     }),
     ignoredBuild: opts.ignoreScripts && shouldBeBuilt,
   }
 }
+
+function getJoinedPath (root: string, sub: string) {
```

Comment:
Why is this function duplicated twice?

Move it to `prepare-package` and let `preparePackage` return the calculated path.

Reply:
Done. To achieve this, I changed the return type of `preparePackage` from `boolean` to `object`.